pallets/utxo/src/lib.rs

```diff
@@ -354,30 +426,79 @@ pub mod pallet {
     #[pallet::getter(fn tokens_higher_id)]
     pub(super) type TokensHigherID<T> = StorageValue<_, TokenID, ValueQuery>;
 
+    /// stores the first block number of every new period
+    #[pallet::storage]
+    #[pallet::getter(fn starting_period)]
+    pub(super) type StartingPeriod<T:Config> = StorageValue<_, T::BlockNumber, ValueQuery>;
+
+    /// stores the extra MLTCoins for dispersing rewards
+    #[pallet::storage]
+    #[pallet::getter(fn coins_available)]
+    pub(super) type MLTCoinsAvailable<T> = StorageValue<_, Value, ValueQuery>;
+
     #[pallet::storage]
     #[pallet::getter(fn reward_total)]
     pub(super) type RewardTotal<T> = StorageValue<_, Value, ValueQuery>;
 
+    #[pallet::storage]
+    #[pallet::getter(fn block_author_reward_amount)]
+    pub(super) type BlockAuthorRewardAmount<T> = StorageValue<_, Value, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn current_block_author)]
+    pub(super) type BlockAuthor<T> = StorageValue<_, Option<H256>, ValueQuery>;
+
     #[pallet::storage]
     #[pallet::getter(fn utxo_store)]
     pub(super) type UtxoStore<T: Config> =
         StorageMap<_, Identity, H256, Option<TransactionOutputFor<T>>, ValueQuery>;
 
+    /// Represents the validators' stakes. When a validator chooses to stop validating,
+    /// the utxo here is transferred back to `UtxoStore`.
+    #[pallet::storage]
+    #[pallet::getter(fn locked_utxos)]
+    pub(super) type LockedUtxos<T: Config> =
+        StorageMap<_, Identity, H256, Option<TransactionOutputFor<T>>, ValueQuery>;
+
+    /// this is to count how many stakings done for that controller account.
+    #[pallet::storage]
+    #[pallet::getter(fn staking_count)]
+    pub(super) type StakingCount<T: Config> = StorageMap<_, Identity,H256,(u64, Value), ValueQuery>;
```

Comment:
Btw, should this save `Option((u64, Value))` instead? I don't actually even know what happens if you call `StakingCount::get()` with a key that doesn't exist. Is garbage returned or does the code panic?

Reply:
changed